docs/querying/datasource.md

```diff
@@ -302,12 +302,14 @@ use join datasources.
 SQL joins take the form:
 
 ```
-<o1> [ INNER | LEFT [OUTER] ] JOIN <o2> ON <condition>
+<o1> [ [ INNER | LEFT [OUTER] JOIN ON ] | CROSS JOIN <o2> 
```

Comment:
Fyi I just remembered, RIGHT [OUTER] join syntax works as well.  Not widely used since SQL tends to read from left to right, but it's in the ANSI standard, and Druid SQL appears to be accepting it:

I'm not sure how the brackets work with "requires one of set" vs "optional", so here are the combinations to merge:

t1 INNER JOIN t2 ON
t1 LEFT [ OUTER ] JOIN  t2 ON
t1 RIGHT [ OUTER ] JOIN  t2 ON
t1 CROSS JOIN  t2

Reply:
Given multiple conditions, I'd go back to the original, simpler format, then use the explanation to say which combinations are not allowed.